library/Vanilla/Utility/ContainerUtils.php

```diff
@@ -67,9 +67,31 @@ public static function currentTheme(): ReferenceInterface {
     public static function cacheBuster(): ReferenceInterface {
         return new Callback(
             function (ContainerInterface $dic) {
-                $cacheBuster = $dic->get(\Vanilla\Web\Asset\DeploymentCacheBuster::class);
+                $cacheBuster = $dic->get(DeploymentCacheBuster::class);
                 return $cacheBuster->value();
             }
         );
     }
+
+    /**
+     * Replace one type of object in the container with another type of object. Existing shared instances will be
+     * overwritten. An alias will be created from the original type to the new type.
+     *
+     * Sometimes an object has limitations or shortcomings that could be resolved by something like a decorator, where
+     * a drop-in replacement wraps existing functionality in enhancements or customizations. This method could configure
+     * a container to use the decorator, replacing stored instances and ensuring new requests to the container would
+     * receive the decorator instead of the original class.
+     *
+     * @param Container $container Container to configure.
+     * @param string $target Container rule to target for replacement. Shared instances will be overwritten.
+     * @param string $replacement Container rule used to determine what will be replace the target in the container.
+     */
+    public static function replace(Container $container, string $target, string $replacement): void {
```

Comment:
I'm going to just leave this as a comment and approve the PR, but...

Do you think `$old` and `$new` are more clear names for this method? I always thought that the needle/haystack were silly paradigms until I realized it was never confusing.